### PR TITLE
Speaking mode

### PR DESCRIPTION
**Pull request by reidash**
Adding of Speaking mode

### CODE REVIEW

File: Project/src/pages/SpeakingMode/SpeakingMode.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+@Component({
+    selector: 'page-SpeakingMode',
+    templateUrl: 'SpeakingMode.html'
+})
+export class SpeakingMode {
+    recognition: any;
+    platform: any;
+    title: string; // Title of the session
+    isCorrect: any;
+    public currUnit: screenUnit; // Current screenUnit
```
Comment (cindyzunic) [inline Project/src/pages/SpeakingMode/SpeakingMode.ts]: Don't think these need to be "public." They're not public in the listening mode file.

File: Project/src/pages/SpeakingMode/SpeakingMode.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                            }
+                        }
+
+                        if (this.isCorrect === true) {
+
+                            this.chooseCorrect();
+                            this.autoAdvance();
+                            return;
+                        }
+						//means out of all the options, non is correct
```
Comment (cindyzunic) [inline Project/src/pages/SpeakingMode/SpeakingMode.ts]: Can you fix the weird spacing here? And the extra new lines below? Sorry, I'm OCD... haha...

File: Project/src/pages/SpeakingMode/SpeakingMode.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+    initUnit = function () {
+        // Logic for setting up a new screenUnit
+        // Maybe add statisitics tracking here?
+        var path = this.plt.is('android') ? cordova.file.applicationDirectory + 'www/' : ''; //might be a hack...
+        this.currState = this.state.init; // Go to initial state
+        this.currUnit = this.screenUnits[this.currIndex]; // Set current screenUnit
+
+        let randomIndex = Math.floor(Math.random() * this.currUnit.audioPaths.length); // Pick audio clip to use
+        this.plt.ready().then((readySource) => { // Make sure the platform is ready before we try to use native components
```
Comment (cindyzunic) [inline Project/src/pages/SpeakingMode/SpeakingMode.ts]: Have a look at ListeningMode.ts. In ListeningMode, I moved this this.plt.ready() check into the constructor. If we leave it in initUnit, we're doing a platform check every time we init a new screen unit, but we don't need to. You can do the same thing here as we do in ListeningMode: 

1. Do the this.plt.ready().... check in the constructor and set a variable "this.loaded = true" when platform is ready.
2. Setup SpeakingMode.html like ListeningMode.html where you have:

```
<ion-content *ngIf="!loaded">
  <div class="loader">Loading...</div>
</ion-content>
```

and then the second <ion-content> (the real content) would have `*ngIf="loaded"`

This way, we only check platform once, and we don't allow the user to interact with anything until platform is ready. Let me know if you need any help!

File: Project/src/pages/SpeakingMode/SpeakingMode.ts (hunk 2 above)
Comment (reidash) [reply Project/src/pages/SpeakingMode/SpeakingMode.ts]: Okay done!

File: Project/src/pages/SpeakingMode/SpeakingMode.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        init: 0,
+        right: 1,
+        wrong: 2,
+        end: 3
+    };
+    public currIndex: number = 0; //index of currently displayed screenUnit
+    public screenUnits: screenUnit[] = []; // Array of all screen units in the session
+
+    constructor(public navCtrl: NavController, public navParams: NavParams, public plt: Platform, private toastCtrl: ToastController, private zone: NgZone) {
+        this.platform = plt;
```
Comment (cindyzunic) [inline Project/src/pages/SpeakingMode/SpeakingMode.ts]: Why are you setting this.platform = plt here but using this.plt below?

**Review by cindyzunic — APPROVED** [review]

**Pull request merged**